src/obis-observer/utils/obis.ts

```diff
@@ -37,3 +52,46 @@ export const toString = ( obis: IObis ): string => {
 
     return result;
 };
+
+
+/**
+ * Convert string to object.
+ *
+ * @param obisString string according to the mask A-B:C.D.E*F
+ */
+export const fromString = ( obisString: string ): IObis => {
+    const result: THashTable = {};
+    let unprocessed = obisString;
+    let parts;
+
+    parts = unprocessed.split('-');
+    if ( parts.length > 1 ) {
+        [result.a, unprocessed] = parts;
+    }
+
+    parts = unprocessed.split(':');
+    if ( parts.length > 1 ) {
+        [result.b, unprocessed] = parts;
+    }
+
+    parts = unprocessed.split('*');
+    if ( parts.length > 1 ) {
+        [unprocessed, result.f] = parts;
+    }
+
+    parts = unprocessed.split('.');
+    if ( parts.length > 1 ) {
+        [result.c, result.d, result.e] = parts;
+    }
+
+    // group value can be number or letter, e.g. 96 or "C"
+    // so need to convert it to proper number
+    Object.keys(result).forEach(groupName => {
+        const groupValue = result[groupName] as string;
+        if ( groupValue ) {
+            result[groupName] = parseInt(letterCodes[groupValue] as string || groupValue, 10);
+        }
+    });
+
+    return result as unknown as IObis;
```

Comment:
```suggestion
    return result as IObis;
```

?

Reply:
nope

Conversion of type 'THashTable' to type 'IObis' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Type 'THashTable' is missing the following properties from type 'IObis': c, dts(2352)